src/nupic/ntypes/Sdr.hpp

```diff
@@ -754,26 +766,99 @@ class SparseDistributedRepresentation : public Serializable
         // Read the dimensions.
         readVector( dimensions_ );
 
-        // Read the data.
-        clear();
-        readVector( flatSparse );
-        flatSparse_valid = true;
-
-        // Consume the end marker.
-        inStream >> marker;
-        NTA_ASSERT( marker == "~SDR" );
-
         // Initialize the SDR.
         // Calculate the SDR's size.
         size_ = 1;
         for(UInt dim : dimensions)
             size_ *= dim;
         // Initialize sparse tuple.
         sparse.assign( dimensions.size(), {} );
+
+        // Read the data.
+        readVector( flatSparse );
+        setFlatSparseInplace();
+
+        // Consume the end marker.
+        inStream >> marker;
+        NTA_ASSERT( marker == "~SDR" );
     };
+
+    void addCallback(SDR_callback_t callback) {
+        callbacks.push_back( callback );
+    }
 };
 
 typedef SparseDistributedRepresentation SDR;
 
+class SDR_Proxy : public SDR
+{
+public:
+    SDR_Proxy(SDR &sdr, const vector<UInt> &dimensions)
+        : SDR( dimensions ) {
+        parent = &sdr;
```

Comment:
do you have to keep parent as member? Could we use copy constructor and call `: SDR(sdr) { ...` ?

Reply:
This needs to be a pointer so that it can update the value of this SDR-Proxy when the parent SDR's value changes.  Otherwise this would get the value of parent once at initialization and never after.